### DISCUSSION TITLE
JsonIgnore on a specific endpoint (Patch vs Update)

### DISCUSSION THREAD

**Discussion by Bartmax** [reactiveui/refit#1640] [Q&A] [answered]
I would like to have 2 patch endpoints, with methos called Patch and Update
both will send the same object but I want to have 1 specific difference,
I want the (`patch`) endpoint to use `JsonIgnoreCondition.WhenWritingNull` while the (`update`) endpoint to have `JsonIgnoreCondition.Never` (the default)

Is this possible with the current library? I only found how to set in the settings on the api.

Thanks,

**Comment by glennawatson** [comment, marked as answer]
JsonIgnoreCondition is a serializer-level setting (RefitSettings.ContentSerializer / JsonSerializerOptions). It applies to the whole client, so there is no per-method override on a single shared model type.

A few ways to get what you want:

1. Two DTO types (cleanest). Give each endpoint its own request type and put the attribute where you want it:

```csharp
public class PatchRequest
{
    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
    public string? Name { get; set; }
}

public class UpdateRequest // default behaviour, nulls written
{
    public string? Name { get; set; }
}

[Patch("/x")] Task Patch([Body] PatchRequest body);
[Patch("/x")] Task Update([Body] UpdateRequest body);
```

The attribute is per-property, so you control it independently per type even with the same JsonSerializerOptions.

2. One model, attribute on the property: put [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)] on the property itself. That is global to the type though, not per-endpoint.

3. Serialize the body yourself and send it as a string/HttpContent for the endpoint that needs different behaviour.

There is no per-method JsonIgnoreCondition switch in Refit; option 1 is the recommended pattern.
